pkg/internal/apis/certmanager/types_certificate.go

```diff
@@ -277,6 +277,12 @@ type CertificateStatus struct {
 	// 1 hour has elapsed from this time.
 	LastFailureTime *metav1.Time
 
+	// LastIssuedTime is the time as recorded by the Certificate controller
+	// of the most recent successfully completed CertificateRequest for this
+	// Certificate resource.
+	// +optional
```

Comment:
nit: We don't need to include `+optional` for our internal API types since they are not code generated, so this directive has no function.

Reply:
> Unfortunately, this is an API change so can't really be backported to v1.1 and so won't fix the bug there. Is there something we can do to not include an API change, or fix the bug separately in v1.1?
This API change seems the cleanest approach to me. Perhaps we implement the following just for v1.1?
re-calculating the actual renewal time (as per the actual certificate in the secret) ...
This introduces the problem where the certificates controller code starts to diverge from later releases which makes it more difficult to backport. Perhaps that's OK. Keen to hear others thoughts.

I will experiment with re-calculating approach a bit whilst waiting to hear what others think.

Reply:
It is possible to implement this with [re-calculating the renewal time in `TriggerController`](https://github.com/irbekrm/cert-manager/commit/9d1a7176848462ba89adb3e85bf712133a2fac95).
It is a less clean approach, but does not require changing the API.
There are some extra comments about this approach from James Munnely on #3603